### DISCUSSION TITLE
PythonAPI on jupyterlab

### DISCUSSION THREAD

**Discussion by JoFeGoOr** [carla-simulator/carla#6803] [Q&A] [answered]
hi all, im new in carla simulator.

i followed the windows build and now i want to write some code following the First steps with CARLA tutorial but i get an error importing carla

on jupyterlab i get a ModuleNotFoundError: No module named 'carla' when i run "import carla" , other imports like import pygame work fine

i have not been able to find a solution online, if someone knows the solution please share

**Comment by Daraan** [comment]
You should have a folder PythonAPI/carla/dist with wheel files use them with `pip install`.
Check if the version there in the filename is matching your python version in the environment else you might need to have a different python environment active during build.

Hope this helps or more information is needed.

**Comment by JoFeGoOr** [comment, marked as answer]
I did not had the \dist forlder, i ran make PythonAPI command twice and it seem to work fine:

C:\carla\PythonAPI\examples>pip install carla
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: carla in c:\users\danielyunge\appdata\roaming\python\python312\site-packages (0.9.14)

**Reply by JoFeGoOr** [reply]
also, i had to install the .whl file in the \dist folder

pip install <wheel-file-name>.whl
